modules/apps/transfer/keeper/relay.go

```diff
@@ -347,7 +347,7 @@ func (k Keeper) refundPacketTokens(ctx sdk.Context, packet channeltypes.Packet,
 			return err
 		}
 
-		if token.Denom.SenderChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel()) {
```

Comment:
I have a feeling that if we are changing to `HasPrefix`, the logic will always be easier to parse if the first if statement is not negated